pkg/generator/ingress_translator.go

```diff
@@ -241,6 +268,52 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 	}, nil
 }
 
+func (translator *IngressTranslator) createUpstreamTransportSocket(activatorCA, activatorSAN string, http2 bool) (*envoy_core_v3.TransportSocket, error) {
+	caSecret, err := translator.secretGetter(system.Namespace(), activatorCA)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch activator CA secret: %w", err)
+	}
+	var alpnProtocols string
+	if http2 {
+		alpnProtocols = "h2"
+	}
+	tlsAny, err := anypb.New(createUpstreamTLSContext(caSecret.Data[caDataName], activatorSAN, alpnProtocols))
+	if err != nil {
+		return nil, err
+	}
+	return &envoy_core_v3.TransportSocket{
+		Name: wellknown.TransportSocketTls,
+		ConfigType: &envoy_core_v3.TransportSocket_TypedConfig{
+			TypedConfig: tlsAny,
+		},
+	}, nil
+}
+
+func createUpstreamTLSContext(caCertificate []byte, activatorSAN string, alpnProtocols ...string) *auth.UpstreamTlsContext {
+	return &auth.UpstreamTlsContext{
+		CommonTlsContext: &auth.CommonTlsContext{
+			AlpnProtocols: alpnProtocols,
+			TlsParams: &auth.TlsParameters{
+				TlsMinimumProtocolVersion: auth.TlsParameters_TLSv1_2,
+			},
+			ValidationContextType: &auth.CommonTlsContext_ValidationContext{
+				ValidationContext: &auth.CertificateValidationContext{
+					TrustedCa: &envoy_core_v3.DataSource{
+						Specifier: &envoy_core_v3.DataSource_InlineBytes{
+							InlineBytes: caCertificate,
+						},
+					},
+					MatchSubjectAltNames: []*envoy_matcher_v3.StringMatcher{{
+						MatchPattern: &envoy_matcher_v3.StringMatcher_Exact{
+							Exact: activatorSAN,
```

Comment:
Could it be the case that ActivatorCa != "" and ActivatorSAN="" ? If so then this part could be omitted?

Reply:
Sorry, I tried to understand but didn't get it. Do you mean we can omit `Exact: activatorSAN`? Then, how to validate the SAN?

Reply:
If ActivatorSAN="" in the configmap but ActivatorCa!= "" does that mean that we still to create the matchPattern? Is there a difference between "" and setting that to nil or completely remove it in that case. Note that if ActivatorSAN!="" then as usually we should the matchPattern here.